File: ChatGPT/trans_test/clojure_to_python/output/output_07b.py
from itertools import product

def parse_line(line):
    lhs, rhs = line.split(": ")
    target = int(lhs)
    nums = list(map(int, rhs.split()))
    return target, nums

def all_operator_combos(n):
    return product(['+', '*', '||'], repeat=n)

def apply_op(a, b, op):
    if op == '+':
        return a + b
    elif op == '*':
        return a * b
    elif op == '||':
        return int(f"{a}{b}")

def eval_left_to_right(nums, ops):
    result = nums[0]
    for op, n in zip(ops, nums[1:]):
        result = apply_op(result, n, op)
    return result

def valid_equation(target, nums):
    for ops in all_operator_combos(len(nums) - 1):
        if eval_left_to_right(nums, ops) == target:
            return True
    return False

def part(lines):
    total = 0
    for line in lines:
        target, nums = parse_line(line)
        if valid_equation(target, nums):
            total += target
    return total

def main():
    with open('input.txt') as f:
        lines = f.read().splitlines()
    print(part(lines))

if __name__ == '__main__':
    main()

File: ChatGPT/trans_test/python_to_clojure/output/output_07b.clj
(ns output_07b
  (:require [clojure.string :as str]))

(defn concat-num [a b]
  (parse-long (str a b)))

(defn eval-expr [nums ops]
  (loop [result (first nums)
         nums (rest nums)
         ops ops]
    (if (empty? ops)
      result
      (let [op (first ops)
            n (first nums)
            new-result (case op
                         "+" (+ result n)
                         "*" (* result n)
                         "||" (concat-num result n))]
        (recur new-result (rest nums) (rest ops))))))

(defn all-op-combinations [n]
  (if (zero? n)
    [[]]
    (for [op ["+" "*" "||"]
          rest (all-op-combinations (dec n))]
      (cons op rest))))

(defn part [lines]
  (reduce
   (fn [total line]
     (if (str/blank? line)
       total
       (let [[lhs rhs] (str/split line #":")
             target (parse-long (str/trim lhs))
             nums (mapv parse-long (str/split (str/trim rhs) #"\s+"))
             ops-list (all-op-combinations (dec (count nums)))]
         (if (some #(= (eval-expr nums %) target) ops-list)
           (+ total target)
           total))))
   0
   lines))

(defn -main [& args]
  (let [filename (first args)
        lines (str/split-lines (slurp filename))]
    (println (part lines))))
